tests/sphere_test.cpp

```diff
@@ -0,0 +1,120 @@
+#include "obj/objects.h"
+#include "gtest/gtest.h"
+
+using core::Ray;
+using core::RayHitData;
+using core::Vec3;
+using objects::Sphere;
+
+const float kSkipFreq = 1000;
+const Vec3 kVecZero(0, 0, 0);
+
+float deg2rad(float deg) { return 2 * constants::kPi * deg / 360; }
+
+TEST(SphereCollisionTest, RayHitFromOutsideSphere) {
+
+  Sphere sphere(kVecZero, 1);
+  RayHitData hitData;
+  Ray alongYAxis(Vec3(0, -4, 0), Vec3(0, 1, 0));
+  ASSERT_TRUE(sphere.hitObject(alongYAxis, kSkipFreq, &hitData));
+  ASSERT_FLOAT_EQ(alongYAxis.origin().magnitude() - sphere.getRadius(),
+                  hitData.time);
+
+  Ray alongYAxisNoHit(Vec3(0, 4, 0), Vec3(0, 1, 0));
+  ASSERT_FALSE(sphere.hitObject(alongYAxisNoHit, kSkipFreq, &hitData))
+      << "hit at: " << hitData.collisionPoint();
+
+  Ray alongXAxis(Vec3(-4, 0, 0), Vec3(1, 0, 0));
```

Comment:
Cases for X and Z are not much useful, given you have Y the same. Remove?

Reply:
removed.